app/controllers/api/expressions_controller.rb

```diff
@@ -0,0 +1,49 @@
+module Api
+  class ExpressionsController < BranchesController
+    rescue_from(ArgumentError) do |_exception|
+      render json: { message: 'unprocessable entity' }, status: :unprocessable_entity
+    end
+    before_action :validate_params
+
+    def show
+      @expression = Expression.new(
+        component: params[:component_id],
+        page: page_with_component
+      )
+
+      render partial: 'expression_answers',
+             locals: {
+               f: default_form_builder.new(:expression, @expression, view_context, {}),
+               conditional_index: params[:conditional_index],
+               expression_index: params[:expression_index]
+             }
+    end
+
+    def page_with_component
+      @page_with_component ||= service.page_with_component(params[:component_id])
+    end
+
+    def component
+      page_with_component.find_component_by_uuid(params[:component_id])
+    end
+
+    delegate :type, to: :component, prefix: true
+    helper_method :component_type
+    def expression_name(type)
+      "branch[conditionals_attributes][#{params[:conditional_index]}]" \
+      "[expressions_attributes][#{params[:expression_index]}][#{type}]"
+    end
+    helper_method :expression_name
+
+    def expression_id(type)
```

Comment:
will `type` call the delegated type above? not sure

Reply:
i don't think so, it renders correctly in the DOM.
But I can see how the naming can be confusing - perhaps I can change the argument from `type` to (** grabs the thesaurus **) `attribute`?